src/ui/webeiddialog.cpp

```diff
@@ -307,8 +304,21 @@ void WebEidDialog::onVerifyPinFailed(const electronic_id::VerifyPinFailed::Statu
     ui->pinErrorLabel->setHidden(message.isEmpty());
     ui->pinErrorLabel->setText(message);
 
-    if (readerHasPinPad.value_or(false)) {
+    if (ui->pinEntryTimeoutProgressBar->isVisible()) {
```

Comment:
You are using the PIN entry timeout progress bar in two roles here, both as a UI widget and as a Boolean variable to indicate the presence of the PIN pad here. This is arguably a clever trick, but I'm afraid that 
- readability suffers by this as this violates the "say what you mean" and "avoid mental mapping" principles,
- this may cause sneaky bugs when you hide the progress bar for some reason, but the PIN pad is still present.

I would vote for keeping explicit `readerHasPinPad`. by rule `#7` "Avoid mental mapping" from the iconic ["Clean Code"](https://mosquitolwz.medium.com/clean-code-quotes-2-meaningful-names-49a3df62121a), but I'm OK with this quite nice trick as well.

Reply:
I don't understand the optional variable purpose. It is or it is not PinPAD. I has or has not progress bar.

Reply:
What you say is exactly its purpose - to say if the reader is or is not a pin PAD. But whether there is a one-to-one correspondence with a progress bar might depend on UI/UX design decisions.

However, this is good enough, so let's move forward with this. Thanks for your work!